test/test_image.py

```diff
@@ -278,5 +279,43 @@ def test_write_file_non_ascii(self):
             os.unlink(fpath)
 
 
+@needs_cuda
+@pytest.mark.parametrize('mode', [ImageReadMode.UNCHANGED, ImageReadMode.GRAY, ImageReadMode.RGB])
+@pytest.mark.parametrize('img_path', get_images(IMAGE_ROOT, ".jpg"))
+@pytest.mark.parametrize('scripted', (False, True))
+def test_decode_jpeg_cuda(mode, img_path, scripted):
+    if 'cmyk' in img_path:
+        pytest.xfail("Decoding a CMYK jpeg isn't supported")
+    tester = ImageTester()
+    data = read_file(img_path)
+    img = decode_image(data, mode=mode)
+    f = torch.jit.script(decode_jpeg) if scripted else decode_jpeg
+    img_nvjpeg = f(data, mode=mode, device='cuda')
+
+    # Some difference expected between jpeg implementations
+    tester.assertTrue((img.float() - img_nvjpeg.cpu().float()).abs().mean() < 2)
+
+
+@needs_cuda
+@pytest.mark.parametrize('cuda_device', ('cuda', 'cuda:0', torch.device('cuda')))
+def test_decode_jpeg_cuda_device_param(cuda_device):
+    """Make sure we can pass a string or a torch.device as device param"""
+    data = read_file(next(get_images(IMAGE_ROOT, ".jpg")))
+    decode_jpeg(data, device=cuda_device)
+
+
+@needs_cuda
+def test_decode_jpeg_cuda_errors():
+    data = read_file(next(get_images(IMAGE_ROOT, ".jpg")))
+    with pytest.raises(RuntimeError, match="Expected a non empty 1-dimensional tensor"):
+        decode_jpeg(data.reshape(-1, 1), device='cuda')
+    with pytest.raises(RuntimeError, match="input tensor must be on CPU"):
+        decode_jpeg(data.to('cuda'), device='cuda')
```

Comment:
What is the reason for requiring the input to be on CPU? Is there a specific limitation on the API of the library? 

If it is possible to have the data on CUDA, then it would be able to call `decode_jpeg()` without specifying a device and assume that the data should remain where they are. This has the benefit of making your methods work exactly like any other operator in TorchVision and potentially might allow you to use the dispatcher (needs to be confirmed).

Reply:
The first stage of nvjpeg decoding is done on the CPU, so it needs to take a CPU buffer as input.

Reply:
Another reason I believe why this takes the input tensor on a CPU device is because we need to get some metadata from the Tensor to allocate the output tensor, which would mean doing a CUDA->CPU copy to perform the allocation.